core/controllers/topic_editor.py

```diff
@@ -0,0 +1,223 @@
+# Copyright 2018 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Controllers for the topics editor, from where topics are edited and stories
+are created.
+"""
+from core.controllers import base
+from core.domain import acl_decorators
+from core.domain import story_domain
+from core.domain import story_services
+from core.domain import topic_domain
+from core.domain import topic_services
+from core.domain import user_services
+import feconf
+import utils
+
+
+def _require_valid_topic_id(topic_id):
+    """Checks whether the topic id received from the frontend is a
+    valid one.
+    """
+    if not isinstance(topic_id, basestring):
+        raise base.BaseHandler.InvalidInputException(
+            Exception('Topic id should be a string.'))
+
+    if len(topic_id) != 12:
+        raise base.BaseHandler.InvalidInputException(
+            Exception('The topic id given is invalid.'))
+
+
+class NewStoryHandler(base.BaseHandler):
+    """Creates a new story."""
+
+    @acl_decorators.can_add_new_story_to_topic
+    def post(self, topic_id):
+        """Handles POST requests."""
```

Comment:
Add some info on "This adds a new canonical story to a topic."

Aim is to distinguish from adding stories in general, which technically could include "additional stories" as well.

Reply:
Done.